pkg/kubelet/metrics/metrics.go

```diff
@@ -25,43 +25,68 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-const kubeletSubsystem = "kubelet"
+const (
+	KubeletSubsystem    = "kubelet"
+	PodWorkerKey        = "pod_worker_latency_microseconds"
+	SyncPodsKey         = "sync_pods_latency_microseconds"
+	PodStartKey         = "pod_start_latency_microseconds"
+	PodStatusKey        = "generate_pod_status_latency_microseconds"
+	DockerOperationsKey = "docker_operations_latency_microseconds"
+	DockerErrorsKey     = "docker_errors"
+)
 
 var (
 	ContainersPerPodCount = prometheus.NewSummary(
 		prometheus.SummaryOpts{
-			Subsystem: kubeletSubsystem,
+			Subsystem: KubeletSubsystem,
 			Name:      "containers_per_pod_count",
 			Help:      "The number of containers per pod.",
 		},
 	)
-	SyncPodLatency = prometheus.NewSummaryVec(
+	PodWorkerLatency = prometheus.NewSummaryVec(
 		prometheus.SummaryOpts{
-			Subsystem: kubeletSubsystem,
-			Name:      "sync_pod_latency_microseconds",
+			Subsystem: KubeletSubsystem,
+			Name:      PodWorkerKey,
 			Help:      "Latency in microseconds to sync a single pod. Broken down by operation type: create, update, or sync",
 		},
-		[]string{"operation_type"},
+		[]string{"operation_type", "pod_name"},
```

Comment:
Oh, I was misled by all the container hashes on my stats page. I remeber having the exact same reservations at some time and finding that we can configure it (http://prometheus.io/docs/operating/configuration/) though I didn't look any deeper. In theory we could just keep around the stats for the last 30m?